Inconsistent segfault in GUI test on Qt signal.emit

https://github.com/Scille/parsec-cloud/runs/7854269358?check_suite_focus=true#step:13:95

```
Run poetry run pytest --log-level=DEBUG --durations=10 --side-effects-timeout=10 -vv -x tests --runmountpoint --runslow --rungui -m gui
[...]
tests/core/gui/test_main_window.py::test_link_file_disconnected_cancel_login PASSED [ 25%]
tests/core/gui/test_main_window.py::test_link_file_unknown_workspace PASSED [ 25%]
tests/core/gui/test_main_window.py::test_link_organization PASSED        [ 25%]
tests/core/gui/test_main_window.py::test_link_organization_disconnected PASSED [ 26%]
Fatal Python error: Segmentation fault

Thread 0x00007fef5dffd700 (most recent call first):
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/trio/_core/_thread_cache.py", line 74 in _work
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 917 in run
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/sentry_sdk/integrations/threading.py", line 67 in run
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 980 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 937 in _bootstrap

Thread 0x00007fef5ffff700 (most recent call first):
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/trio/_core/_thread_cache.py", line 74 in _work
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 917 in run
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/sentry_sdk/integrations/threading.py", line 67 in run
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 980 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 937 in _bootstrap

Thread 0x00007fef91dfe700 (most recent call first):
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/trio/_core/_thread_cache.py", line 74 in _work
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 917 in run
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/sentry_sdk/integrations/threading.py", line 67 in run
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 980 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 937 in _bootstrap

Thread 0x00007fef92f20700 (most recent call first):
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/trio/_core/_thread_cache.py", line 74 in _work
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 917 in run
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/sentry_sdk/integrations/threading.py", line 67 in run
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 980 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 937 in _bootstrap

Thread 0x00007fef93fff700 (most recent call first):
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/trio/_core/_thread_cache.py", line 74 in _work
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 917 in run
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 980 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 937 in _bootstrap

Thread 0x00007fefb56a1700 (most recent call first):
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/socket.py", line 293 in accept
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pytest_rerunfailures.py", line 429 in run_server
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 917 in run
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 980 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 937 in _bootstrap

Current thread 0x00007fefba7b7740 (most recent call first):
  File "/home/runner/work/parsec-cloud/parsec-cloud/parsec/core/gui/trio_jobs.py", line 127 in _set_done
  File "/home/runner/work/parsec-cloud/parsec-cloud/parsec/core/gui/trio_jobs.py", line 106 in set_cancelled
  File "/home/runner/work/parsec-cloud/parsec-cloud/parsec/core/gui/trio_jobs.py", line 86 in __call__
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/trio/_core/_run.py", line 2181 in unrolled_run
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/outcome/_impl.py", line 113 in send
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/trio/_core/_run.py", line 1242 in guest_tick
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/qtrio/qt.py", line 31 in event
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/qtrio/_core.py", line 652 in run
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/qtrio/_core.py", line 510 in run
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pytest_trio/plugin.py", line 341 in wrapper
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/python.py", line 192 in pytest_pyfunc_call
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/python.py", line 1761 in runtest
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/runner.py", line 166 in pytest_runtest_call
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/runner.py", line 259 in <lambda>
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/runner.py", line 338 in from_call
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/runner.py", line 258 in call_runtest_hook
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/runner.py", line 219 in call_and_report
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/runner.py", line 130 in runtestprotocol
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/runner.py", line 111 in pytest_runtest_protocol
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/main.py", line 347 in pytest_runtestloop
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/main.py", line 322 in _main
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/main.py", line 268 in wrap_session
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/main.py", line 315 in pytest_cmdline_main
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/config/__init__.py", line 164 in main
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/config/__init__.py", line 187 in console_main
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/bin/pytest", line 8 in <module>
/home/runner/work/_temp/355b9315-4806-4c63-be85-a45138d4221b.sh: line 1: 13488 Segmentation fault      (core dumped) poetry run pytest --log-level=DEBUG --durations=10 --side-effects-timeout=10 -vv -x tests --runmountpoint --runslow --rungui -m gui
tests/core/gui/test_main_window.py::test_link_claim_device[True] 
Error: Process completed with exit code 139.
```

another one: https://github.com/Scille/parsec-cloud/runs/8068077649?check_suite_focus=true

```
tests/core/gui/test_main_window.py::test_link_claim_device[True] PASSED  [ 27%]
Fatal Python error: Segmentation fault

Thread 0x00007f5512ffe700 (most recent call first):
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/trio/_core/_thread_cache.py", line 74 in _work
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 917 in run
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/sentry_sdk/integrations/threading.py", line 67 in run
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 980 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 937 in _bootstrap

Thread 0x00007f5521ffd700 (most recent call first):
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/trio/_core/_thread_cache.py", line 74 in _work
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 917 in run
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/sentry_sdk/integrations/threading.py", line 67 in run
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 980 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 937 in _bootstrap

Thread 0x00007f553d848700 (most recent call first):
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/trio/_core/_thread_cache.py", line 74 in _work
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 917 in run
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/sentry_sdk/integrations/threading.py", line 67 in run
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 980 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 937 in _bootstrap

Thread 0x00007f553e849700 (most recent call first):
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/trio/_core/_thread_cache.py", line 74 in _work
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 917 in run
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/sentry_sdk/integrations/threading.py", line 67 in run
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 980 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 937 in _bootstrap

Thread 0x00007f5551473700 (most recent call first):
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/trio/_core/_thread_cache.py", line 74 in _work
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 917 in run
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 980 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 937 in _bootstrap

Thread 0x00007f556ee56700 (most recent call first):
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/socket.py", line 293 in accept
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pytest_rerunfailures.py", line 429 in run_server
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 917 in run
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 980 in _bootstrap_inner
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/threading.py", line 937 in _bootstrap

Current thread 0x00007f55740b1740 (most recent call first):
  File "/home/runner/work/parsec-cloud/parsec-cloud/parsec/core/gui/trio_jobs.py", line 127 in _set_done
  File "/home/runner/work/parsec-cloud/parsec-cloud/parsec/core/gui/trio_jobs.py", line 106 in set_cancelled
  File "/home/runner/work/parsec-cloud/parsec-cloud/parsec/core/gui/trio_jobs.py", line 86 in __call__
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/trio/_core/_run.py", line 2181 in unrolled_run
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/outcome/_impl.py", line 113 in send
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/trio/_core/_run.py", line 1242 in guest_tick
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/qtrio/qt.py", line 31 in event
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/qtrio/_core.py", line 652 in run
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/qtrio/_core.py", line 510 in run
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pytest_trio/plugin.py", line 341 in wrapper
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/python.py", line 192 in pytest_pyfunc_call
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/python.py", line 1761 in runtest
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/runner.py", line 166 in pytest_runtest_call
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/runner.py", line 259 in <lambda>
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/runner.py", line 338 in from_call
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/runner.py", line 258 in call_runtest_hook
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/runner.py", line 219 in call_and_report
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/runner.py", line 130 in runtestprotocol
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/runner.py", line 111 in pytest_runtest_protocol
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/main.py", line 347 in pytest_runtestloop
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/main.py", line 322 in _main
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/main.py", line 268 in wrap_session
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/main.py", line 315 in pytest_cmdline_main
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/config/__init__.py", line 164 in main
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/lib/python3.9/site-packages/_pytest/config/__init__.py", line 187 in console_main
  File "/home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/bin/pytest", line 8 in <module>
/home/runner/work/_temp/3a927ded-ac21-4870-82a3-030aea682465.sh: line 1: 19331 Segmentation fault      (core dumped) poetry run pytest --log-level=DEBUG --durations=10 --side-effects-timeout=10 -vv -x tests --runmountpoint --runslow --rungui -m gui
tests/core/gui/test_main_window.py::test_link_claim_device[False] 
Error: Process completed with exit code 139.
```

--- a/newsfragments/2839.bugfix.rst
+++ b/newsfragments/2839.bugfix.rst
@@ -1,0 +1,1 @@
+Fix rare segfault in the GUI when an event occurs at the wrong time

--- a/parsec/core/gui/claim_device_widget.py
+++ b/parsec/core/gui/claim_device_widget.py
@@ -225,13 +225,17 @@ class ClaimDeviceCodeExchangeWidget(QWidget, Ui_ClaimDeviceCodeExchangeWidget):
         self.wait_peer_trust_error.connect(self._on_wait_peer_trust_error)
 
         self.get_greeter_sas_job = self.jobs_ctx.submit_job(
-            self.get_greeter_sas_success, self.get_greeter_sas_error, self.claimer.get_greeter_sas
+            (self, "get_greeter_sas_success"),
+            (self, "get_greeter_sas_error"),
+            self.claimer.get_greeter_sas,
         )
 
     def _on_good_greeter_code_clicked(self):
         self.widget_greeter_code.setDisabled(True)
         self.signify_trust_job = self.jobs_ctx.submit_job(
-            self.signify_trust_success, self.signify_trust_error, self.claimer.signify_trust
+            (self, "signify_trust_success"),
+            (self, "signify_trust_error"),
+            self.claimer.signify_trust,
         )
 
     def _on_wrong_greeter_code_clicked(self):
@@ -283,7 +287,9 @@ class ClaimDeviceCodeExchangeWidget(QWidget, Ui_ClaimDeviceCodeExchangeWidget):
         self.widget_claimer_code.setVisible(True)
         self.line_edit_claimer_code.setText(claimer_sas.str)
         self.wait_peer_trust_job = self.jobs_ctx.submit_job(
-            self.wait_peer_trust_success, self.wait_peer_trust_error, self.claimer.wait_peer_trust
+            (self, "wait_peer_trust_success"),
+            (self, "wait_peer_trust_error"),
+            self.claimer.wait_peer_trust,
         )
 
     def _on_get_claimer_sas_error(self, job):
@@ -303,7 +309,9 @@ class ClaimDeviceCodeExchangeWidget(QWidget, Ui_ClaimDeviceCodeExchangeWidget):
         assert job.is_finished()
         assert job.status == "ok"
         self.get_claimer_sas_job = self.jobs_ctx.submit_job(
-            self.get_claimer_sas_success, self.get_claimer_sas_error, self.claimer.get_claimer_sas
+            (self, "get_claimer_sas_success"),
+            (self, "get_claimer_sas_error"),
+            self.claimer.get_claimer_sas,
         )
 
     def _on_signify_trust_error(self, job):
@@ -402,8 +410,8 @@ class ClaimDeviceProvideInfoWidget(QWidget, Ui_ClaimDeviceProvideInfoWidget):
             self.widget_info.setDisabled(True)
             self.label_wait.show()
             self.claim_job = self.jobs_ctx.submit_job(
-                self.claim_success,
-                self.claim_error,
+                (self, "claim_success"),
+                (self, "claim_error"),
                 self.claimer.claim_device,
                 device_label=device_label,
             )
@@ -474,7 +482,7 @@ class ClaimDeviceInstructionsWidget(QWidget, Ui_ClaimDeviceInstructionsWidget):
         self.button_start.setDisabled(True)
         self.button_start.setText(_("TEXT_CLAIM_DEVICE_WAITING"))
         self.wait_peer_job = self.jobs_ctx.submit_job(
-            self.wait_peer_success, self.wait_peer_error, self.claimer.wait_peer
+            (self, "wait_peer_success"), (self, "wait_peer_error"), self.claimer.wait_peer
         )
 
     def _on_wait_peer_success(self, job):
@@ -543,14 +551,16 @@ class ClaimDeviceWidget(QWidget, Ui_ClaimDeviceWidget):
 
     def _run_claimer(self):
         self.claimer_job = self.jobs_ctx.submit_job(
-            self.claimer_success,
-            self.claimer_error,
+            (self, "claimer_success"),
+            (self, "claimer_error"),
             self.claimer.run,
             addr=self.addr,
             config=self.config,
         )
         self.retrieve_info_job = self.jobs_ctx.submit_job(
-            self.retrieve_info_success, self.retrieve_info_error, self.claimer.retrieve_info
+            (self, "retrieve_info_success"),
+            (self, "retrieve_info_error"),
+            self.claimer.retrieve_info,
         )
 
     def _on_retrieve_info_success(self, job):

--- a/parsec/core/gui/claim_user_widget.py
+++ b/parsec/core/gui/claim_user_widget.py
@@ -294,13 +294,17 @@ class ClaimUserCodeExchangeWidget(QWidget, Ui_ClaimUserCodeExchangeWidget):
         self.wait_peer_trust_error.connect(self._on_wait_peer_trust_error)
 
         self.get_greeter_sas_job = self.jobs_ctx.submit_job(
-            self.get_greeter_sas_success, self.get_greeter_sas_error, self.claimer.get_greeter_sas
+            (self, "get_greeter_sas_success"),
+            (self, "get_greeter_sas_error"),
+            self.claimer.get_greeter_sas,
         )
 
     def _on_good_greeter_code_clicked(self):
         self.widget_greeter_code.setDisabled(True)
         self.signify_trust_job = self.jobs_ctx.submit_job(
-            self.signify_trust_success, self.signify_trust_error, self.claimer.signify_trust
+            (self, "signify_trust_success"),
+            (self, "signify_trust_error"),
+            self.claimer.signify_trust,
         )
 
     def _on_wrong_greeter_code_clicked(self):
@@ -350,7 +354,9 @@ class ClaimUserCodeExchangeWidget(QWidget, Ui_ClaimUserCodeExchangeWidget):
         self.widget_claimer_code.setVisible(True)
         self.line_edit_claimer_code.setText(claimer_sas.str)
         self.wait_peer_trust_job = self.jobs_ctx.submit_job(
-            self.wait_peer_trust_success, self.wait_peer_trust_error, self.claimer.wait_peer_trust
+            (self, "wait_peer_trust_success"),
+            (self, "wait_peer_trust_error"),
+            self.claimer.wait_peer_trust,
         )
 
     def _on_get_claimer_sas_error(self, job):
@@ -368,7 +374,9 @@ class ClaimUserCodeExchangeWidget(QWidget, Ui_ClaimUserCodeExchangeWidget):
         assert self.signify_trust_job.status == "ok"
         self.signify_trust_job = None
         self.get_claimer_sas_job = self.jobs_ctx.submit_job(
-            self.get_claimer_sas_success, self.get_claimer_sas_error, self.claimer.get_claimer_sas
+            (self, "get_claimer_sas_success"),
+            (self, "get_claimer_sas_error"),
+            self.claimer.get_claimer_sas,
         )
 
     def _on_signify_trust_error(self, job):
@@ -482,8 +490,8 @@ class ClaimUserProvideInfoWidget(QWidget, Ui_ClaimUserProvideInfoWidget):
         self.widget_info.setDisabled(True)
         self.label_wait.show()
         self.claim_job = self.jobs_ctx.submit_job(
-            self.claim_success,
-            self.claim_error,
+            (self, "claim_success"),
+            (self, "claim_error"),
             self.claimer.claim_user,
             device_label=device_label,
             human_handle=human_handle,
@@ -553,7 +561,7 @@ class ClaimUserInstructionsWidget(QWidget, Ui_ClaimUserInstructionsWidget):
         self.button_start.setDisabled(True)
         self.button_start.setText(_("TEXT_CLAIM_USER_WAITING"))
         self.wait_peer_job = self.jobs_ctx.submit_job(
-            self.wait_peer_success, self.wait_peer_error, self.claimer.wait_peer
+            (self, "wait_peer_success"), (self, "wait_peer_error"), self.claimer.wait_peer
         )
 
     def _on_wait_peer_success(self, job):
@@ -618,14 +626,16 @@ class ClaimUserWidget(QWidget, Ui_ClaimUserWidget):
 
     def _run_claimer(self):
         self.claimer_job = self.jobs_ctx.submit_job(
-            self.claimer_success,
-            self.claimer_error,
+            (self, "claimer_success"),
+            (self, "claimer_error"),
             self.claimer.run,
             addr=self.addr,
             config=self.config,
         )
         self.retrieve_info_job = self.jobs_ctx.submit_job(
-            self.retrieve_info_success, self.retrieve_info_error, self.claimer.retrieve_info
+            (self, "retrieve_info_success"),
+            (self, "retrieve_info_error"),
+            self.claimer.retrieve_info,
         )
 
     def _on_retrieve_info_success(self, job):

--- a/parsec/core/gui/create_org_widget.py
+++ b/parsec/core/gui/create_org_widget.py
@@ -218,8 +218,8 @@ class CreateOrgWidget(QWidget, Ui_CreateOrgWidget):
 
             # TODO: call `await _do_create_org` directly since the context is now async
             self.create_job = self.jobs_ctx.submit_job(
-                self.req_success,
-                self.req_error,
+                (self, "req_success"),
+                (self, "req_error"),
                 _do_create_org,
                 config=self.config,
                 human_handle=human_handle,

--- a/parsec/core/gui/device_recovery_export_widget.py
+++ b/parsec/core/gui/device_recovery_export_widget.py
@@ -205,8 +205,8 @@ class DeviceRecoveryExportWidget(QWidget, Ui_DeviceRecoveryExportWidget):
                     self.button_validate.setEnabled(True)
                     return
             self.jobs_ctx.submit_job(
-                self.export_success,
-                self.export_failure,
+                (self, "export_success"),
+                (self, "export_failure"),
                 self._export_recovery_device,
                 config_dir=self.config.config_dir,
                 device=device,

--- a/parsec/core/gui/device_recovery_import_widget.py
+++ b/parsec/core/gui/device_recovery_import_widget.py
@@ -140,8 +140,8 @@ class DeviceRecoveryImportWidget(QWidget, Ui_DeviceRecoveryImportWidget):
             # No try/except given `self.line_edit_device` has already been validated against `DeviceLabel`
             device_label = DeviceLabel(self.current_page.line_edit_device.clean_text())
             self.jobs_ctx.submit_job(
-                self.create_new_device_success,
-                self.create_new_device_failure,
+                (self, "create_new_device_success"),
+                (self, "create_new_device_failure"),
                 self._create_new_device,
                 device_label=device_label,
                 file_path=PurePath(self.current_page.get_recovery_key_file()),

--- a/parsec/core/gui/devices_widget.py
+++ b/parsec/core/gui/devices_widget.py
@@ -85,7 +85,7 @@ class DevicesWidget(QWidget, Ui_DevicesWidget):
 
     def invite_device(self):
         self.jobs_ctx.submit_job(
-            self.invite_success, self.invite_error, _do_invite_device, core=self.core
+            (self, "invite_success"), (self, "invite_error"), _do_invite_device, core=self.core
         )
 
     def _on_invite_success(self, job):
@@ -145,5 +145,5 @@ class DevicesWidget(QWidget, Ui_DevicesWidget):
         self.layout_devices.clear()
         self.spinner.show()
         self.jobs_ctx.submit_job(
-            self.list_success, self.list_error, _do_list_devices, core=self.core
+            (self, "list_success"), (self, "list_error"), _do_list_devices, core=self.core
         )

--- a/parsec/core/gui/enrollment_widget.py
+++ b/parsec/core/gui/enrollment_widget.py
@@ -223,7 +223,7 @@ class EnrollmentWidget(QWidget, Ui_EnrollmentWidget):
         if self.current_job is not None and not self.current_job.is_finished():
             return
         self.current_job = self.jobs_ctx.submit_job(
-            self.list_success, self.list_failure, self.list_pending_enrollments
+            (self, "list_success"), (self, "list_failure"), self.list_pending_enrollments
         )
 
     def clear_layout(self):

--- a/parsec/core/gui/file_history_widget.py
+++ b/parsec/core/gui/file_history_widget.py
@@ -102,8 +102,8 @@ class FileHistoryWidget(QWidget, Ui_FileHistoryWidget):
                 w.hide()
                 w.setParent(0)
         self.versions_job = self.jobs_ctx.submit_job(
-            self.get_versions_success,
-            self.get_versions_error,
+            (self, "get_versions_success"),
+            (self, "get_versions_error"),
             _do_workspace_version,
             version_lister=self.version_lister,
             path=self.path,

--- a/parsec/core/gui/file_status_widget.py
+++ b/parsec/core/gui/file_status_widget.py
@@ -26,7 +26,9 @@ class FileStatusWidget(QWidget, Ui_FileInfoWidget):
         self.path = path
         self.core = core
 
-        self.jobs_ctx.submit_job(self.get_status_success, self.get_status_error, self.get_status)
+        self.jobs_ctx.submit_job(
+            (self, "get_status_success"), (self, "get_status_error"), self.get_status
+        )
 
     async def get_status(self):
         path_info = await self.workspace_fs.path_info(self.path)

--- a/parsec/core/gui/files_widget.py
+++ b/parsec/core/gui/files_widget.py
@@ -447,8 +447,8 @@ class FilesWidget(QWidget, Ui_FilesWidget):
 
         if self.clipboard.status == Clipboard.Status.Cut:
             self.jobs_ctx.submit_job(
-                self.move_success,
-                self.move_error,
+                (self, "move_success"),
+                (self, "move_error"),
                 _do_move_files,
                 workspace_fs=self.workspace_fs,
                 target_dir=self.current_directory,
@@ -462,8 +462,8 @@ class FilesWidget(QWidget, Ui_FilesWidget):
 
         elif self.clipboard.status == Clipboard.Status.Copied:
             self.jobs_ctx.submit_job(
-                self.copy_success,
-                self.copy_error,
+                (self, "copy_success"),
+                (self, "copy_error"),
                 _do_copy_files,
                 workspace_fs=self.workspace_fs,
                 target_dir=self.current_directory,
@@ -559,8 +559,8 @@ class FilesWidget(QWidget, Ui_FilesWidget):
             if not new_name:
                 return
             self.jobs_ctx.submit_job(
-                self.rename_success,
-                self.rename_error,
+                (self, "rename_success"),
+                (self, "rename_error"),
                 _do_rename,
                 workspace_fs=self.workspace_fs,
                 paths=[
@@ -584,8 +584,8 @@ class FilesWidget(QWidget, Ui_FilesWidget):
                 return
 
             self.jobs_ctx.submit_job(
-                self.rename_success,
-                self.rename_error,
+                (self, "rename_success"),
+                (self, "rename_error"),
                 _do_rename,
                 workspace_fs=self.workspace_fs,
                 paths=[
@@ -618,8 +618,8 @@ class FilesWidget(QWidget, Ui_FilesWidget):
         if result != _("ACTION_FILE_DELETE_MULTIPLE") and result != _("ACTION_FILE_DELETE"):
             return
         self.jobs_ctx.submit_job(
-            self.delete_success,
-            self.delete_error,
+            (self, "delete_success"),
+            (self, "delete_error"),
             _do_delete,
             workspace_fs=self.workspace_fs,
             files=[(self.current_directory / f.name, f.type) for f in files],
@@ -641,7 +641,7 @@ class FilesWidget(QWidget, Ui_FilesWidget):
             for name in names
         ]
         self.jobs_ctx.submit_job(
-            self.file_open_success, self.file_open_error, desktop.open_files_job, paths
+            (self, "file_open_success"), (self, "file_open_error"), desktop.open_files_job, paths
         )
 
     def open_files(self):
@@ -684,8 +684,8 @@ class FilesWidget(QWidget, Ui_FilesWidget):
         self.jobs_ctx.submit_throttled_job(
             "files_widget.reload",
             delay,
-            self.folder_stat_success,
-            self.folder_stat_error,
+            (self, "folder_stat_success"),
+            (self, "folder_stat_error"),
             _do_folder_stat,
             workspace_fs=self.workspace_fs,
             path=self.current_directory,
@@ -699,8 +699,8 @@ class FilesWidget(QWidget, Ui_FilesWidget):
         self.current_directory = directory
         self.current_directory_id = None
         self.jobs_ctx.submit_job(
-            self.folder_stat_success,
-            self.folder_stat_error,
+            (self, "folder_stat_success"),
+            (self, "folder_stat_error"),
             _do_folder_stat,
             workspace_fs=self.workspace_fs,
             path=directory,
@@ -717,7 +717,11 @@ class FilesWidget(QWidget, Ui_FilesWidget):
             return
         self.default_import_path = str(pathlib.Path(paths[0]).parent)
         self.jobs_ctx.submit_job(
-            self.import_success, self.import_error, self._do_import, paths, self.current_directory
+            (self, "import_success"),
+            (self, "import_error"),
+            self._do_import,
+            paths,
+            self.current_directory,
         )
 
     def import_folder_clicked(self):
@@ -728,7 +732,11 @@ class FilesWidget(QWidget, Ui_FilesWidget):
             return
         self.default_import_path = str(path)
         self.jobs_ctx.submit_job(
-            self.import_success, self.import_error, self._do_import, [path], self.current_directory
+            (self, "import_success"),
+            (self, "import_error"),
+            self._do_import,
+            [path],
+            self.current_directory,
         )
 
     def on_files_dropped(self, sources, dest):
@@ -739,7 +747,7 @@ class FilesWidget(QWidget, Ui_FilesWidget):
         else:
             dest = self.current_directory / dest
         self.jobs_ctx.submit_job(
-            self.import_success, self.import_error, self._do_import, sources, dest
+            (self, "import_success"), (self, "import_error"), self._do_import, sources, dest
         )
 
     # Import job
@@ -949,8 +957,8 @@ class FilesWidget(QWidget, Ui_FilesWidget):
         else:
             target_dir = self.current_directory / target_name
         self.jobs_ctx.submit_job(
-            self.move_success,
-            self.move_error,
+            (self, "move_success"),
+            (self, "move_error"),
             _do_move_files,
             workspace_fs=self.workspace_fs,
             target_dir=target_dir,
@@ -982,8 +990,8 @@ class FilesWidget(QWidget, Ui_FilesWidget):
             return
 
         self.jobs_ctx.submit_job(
-            self.folder_create_success,
-            self.folder_create_error,
+            (self, "folder_create_success"),
+            (self, "folder_create_error"),
             _do_folder_create,
             workspace_fs=self.workspace_fs,
             path=self.current_directory / folder_name,
@@ -1185,8 +1193,8 @@ class FilesWidget(QWidget, Ui_FilesWidget):
         self, timestamp, path, file_type, open_after_load, close_after_remount, reload_after_remount
     ):
         self.jobs_ctx.submit_job(
-            self.reload_timestamped_success,
-            self.reload_timestamped_error,
+            (self, "reload_timestamped_success"),
+            (self, "reload_timestamped_error"),
             _do_remount_timestamped,
             mountpoint_manager=self.core.mountpoint_manager,
             workspace_fs=self.workspace_fs,

--- a/parsec/core/gui/greet_device_widget.py
+++ b/parsec/core/gui/greet_device_widget.py
@@ -173,7 +173,7 @@ class GreetDeviceInstructionsWidget(QWidget, Ui_GreetDeviceInstructionsWidget):
     def _on_button_send_email_clicked(self):
         self.button_send_email.setDisabled(True)
         self.jobs_ctx.submit_job(
-            self.send_email_success, self.send_email_error, _do_send_email, core=self.core
+            (self, "send_email_success"), (self, "send_email_error"), _do_send_email, core=self.core
         )
 
     def _on_send_email_success(self, job):
@@ -221,7 +221,7 @@ class GreetDeviceInstructionsWidget(QWidget, Ui_GreetDeviceInstructionsWidget):
         self.button_send_email.setDisabled(True)
         self.button_start.setText(_("TEXT_GREET_DEVICE_WAITING"))
         self.wait_peer_job = self.jobs_ctx.submit_job(
-            self.wait_peer_success, self.wait_peer_error, self.greeter.wait_peer
+            (self, "wait_peer_success"), (self, "wait_peer_error"), self.greeter.wait_peer
         )
 
     def _on_wait_peer_success(self, job):
@@ -304,14 +304,18 @@ class GreetDeviceCodeExchangeWidget(QWidget, Ui_GreetDeviceCodeExchangeWidget):
         self.label_wait_info.hide()
 
         self.get_greeter_sas_job = self.jobs_ctx.submit_job(
-            self.get_greeter_sas_success, self.get_greeter_sas_error, self.greeter.get_greeter_sas
+            (self, "get_greeter_sas_success"),
+            (self, "get_greeter_sas_error"),
+            self.greeter.get_greeter_sas,
         )
 
     def _on_good_claimer_code_clicked(self):
         self.widget_claimer_code.hide()
         self.label_wait_info.show()
         self.signify_trust_job = self.jobs_ctx.submit_job(
-            self.signify_trust_success, self.signify_trust_error, self.greeter.signify_trust
+            (self, "signify_trust_success"),
+            (self, "signify_trust_error"),
+            self.greeter.signify_trust,
         )
 
     def _on_wrong_claimer_code_clicked(self):
@@ -332,7 +336,9 @@ class GreetDeviceCodeExchangeWidget(QWidget, Ui_GreetDeviceCodeExchangeWidget):
         greeter_sas = job.ret
         self.line_edit_greeter_code.setText(greeter_sas.str)
         self.wait_peer_trust_job = self.jobs_ctx.submit_job(
-            self.wait_peer_trust_success, self.wait_peer_trust_error, self.greeter.wait_peer_trust
+            (self, "wait_peer_trust_success"),
+            (self, "wait_peer_trust_error"),
+            self.greeter.wait_peer_trust,
         )
 
     def _on_get_greeter_sas_error(self, job):
@@ -421,7 +427,9 @@ class GreetDeviceCodeExchangeWidget(QWidget, Ui_GreetDeviceCodeExchangeWidget):
         assert job.is_finished()
         assert job.status == "ok"
         self.get_claimer_sas_job = self.jobs_ctx.submit_job(
-            self.get_claimer_sas_success, self.get_claimer_sas_error, self.greeter.get_claimer_sas
+            (self, "get_claimer_sas_success"),
+            (self, "get_claimer_sas_error"),
+            self.greeter.get_claimer_sas,
         )
 
     def _on_wait_peer_trust_error(self, job):
@@ -463,8 +471,8 @@ class GreetDeviceWidget(QWidget, Ui_GreetDeviceWidget):
 
     def _run_greeter(self):
         self.greeter_job = self.jobs_ctx.submit_job(
-            self.greeter_success,
-            self.greeter_error,
+            (self, "greeter_success"),
+            (self, "greeter_error"),
             self.greeter.run,
             core=self.core,
             token=self.invite_addr.token,

--- a/parsec/core/gui/greet_user_widget.py
+++ b/parsec/core/gui/greet_user_widget.py
@@ -185,7 +185,7 @@ class GreetUserInstructionsWidget(QWidget, Ui_GreetUserInstructionsWidget):
         self.button_start.setDisabled(True)
         self.button_start.setText(_("TEXT_GREET_USER_WAITING"))
         self.wait_peer_job = self.jobs_ctx.submit_job(
-            self.wait_peer_success, self.wait_peer_error, self.greeter.wait_peer
+            (self, "wait_peer_success"), (self, "wait_peer_error"), self.greeter.wait_peer
         )
 
     def _on_wait_peer_success(self, job):
@@ -275,7 +275,9 @@ class GreetUserCheckInfoWidget(QWidget, Ui_GreetUserCheckInfoWidget):
         self.button_create_user.clicked.connect(self._on_create_user_clicked)
 
         self.get_requests_job = self.jobs_ctx.submit_job(
-            self.get_requests_success, self.get_requests_error, self.greeter.get_claim_requests
+            (self, "get_requests_success"),
+            (self, "get_requests_error"),
+            self.greeter.get_claim_requests,
         )
 
     def check_infos(self, _=None):
@@ -300,8 +302,8 @@ class GreetUserCheckInfoWidget(QWidget, Ui_GreetUserCheckInfoWidget):
         self.button_create_user.setDisabled(True)
         self.button_create_user.setText(_("TEXT_GREET_USER_WAITING"))
         self.create_user_job = self.jobs_ctx.submit_job(
-            self.create_user_success,
-            self.create_user_error,
+            (self, "create_user_success"),
+            (self, "create_user_error"),
             self.greeter.create_new_user,
             human_handle=handle,
             device_label=device_label,
@@ -421,13 +423,17 @@ class GreetUserCodeExchangeWidget(QWidget, Ui_GreetUserCodeExchangeWidget):
         self.get_claimer_sas_error.connect(self._on_get_claimer_sas_error)
 
         self.get_greeter_sas_job = self.jobs_ctx.submit_job(
-            self.get_greeter_sas_success, self.get_greeter_sas_error, self.greeter.get_greeter_sas
+            (self, "get_greeter_sas_success"),
+            (self, "get_greeter_sas_error"),
+            self.greeter.get_greeter_sas,
         )
 
     def _on_good_claimer_code_clicked(self):
         self.widget_claimer_code.setDisabled(True)
         self.signify_trust_job = self.jobs_ctx.submit_job(
-            self.signify_trust_success, self.signify_trust_error, self.greeter.signify_trust
+            (self, "signify_trust_success"),
+            (self, "signify_trust_error"),
+            self.greeter.signify_trust,
         )
 
     def _on_wrong_claimer_code_clicked(self):
@@ -448,7 +454,9 @@ class GreetUserCodeExchangeWidget(QWidget, Ui_GreetUserCodeExchangeWidget):
         greeter_sas = job.ret
         self.line_edit_greeter_code.setText(greeter_sas.str)
         self.wait_peer_trust_job = self.jobs_ctx.submit_job(
-            self.wait_peer_trust_success, self.wait_peer_trust_error, self.greeter.wait_peer_trust
+            (self, "wait_peer_trust_success"),
+            (self, "wait_peer_trust_error"),
+            self.greeter.wait_peer_trust,
         )
 
     def _on_get_greeter_sas_error(self, job):
@@ -537,7 +545,9 @@ class GreetUserCodeExchangeWidget(QWidget, Ui_GreetUserCodeExchangeWidget):
         assert job.is_finished()
         assert job.status == "ok"
         self.get_claimer_sas_job = self.jobs_ctx.submit_job(
-            self.get_claimer_sas_success, self.get_claimer_sas_error, self.greeter.get_claimer_sas
+            (self, "get_claimer_sas_success"),
+            (self, "get_claimer_sas_error"),
+            self.greeter.get_claimer_sas,
         )
 
     def _on_wait_peer_trust_error(self, job):
@@ -579,8 +589,8 @@ class GreetUserWidget(QWidget, Ui_GreetUserWidget):
 
     def _run_greeter(self):
         self.greeter_job = self.jobs_ctx.submit_job(
-            self.greeter_success,
-            self.greeter_error,
+            (self, "greeter_success"),
+            (self, "greeter_error"),
             self.greeter.run,
             core=self.core,
             token=self.token,

--- a/parsec/core/gui/instance_widget.py
+++ b/parsec/core/gui/instance_widget.py
@@ -1,9 +1,9 @@
 # Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
 
 import trio
-from typing import Optional
+from typing import Optional, Tuple
 from structlog import get_logger
-from PyQt5.QtCore import pyqtSignal, pyqtBoundSignal
+from PyQt5.QtCore import QObject, pyqtSignal, pyqtBoundSignal
 from PyQt5.QtWidgets import QWidget, QVBoxLayout, QApplication
 from packaging.version import Version
 
@@ -39,7 +39,8 @@ logger = get_logger()
 MIN_MACFUSE_VERSION = Version("4.4.0")
 
 
-async def _do_run_core(config, device, qt_on_ready):
+async def _do_run_core(config, device, qt_on_ready: Tuple[QObject, str]):
+    qt_on_ready: pyqtBoundSignal = getattr(qt_on_ready[0], qt_on_ready[1])  # Retreive the signal
     # Quick fix to avoid MultiError<Cancelled, ...> exception bubbling up
     # TODO: replace this by a proper generic MultiError handling
     with trio.MultiError.catch(lambda exc: None if isinstance(exc, trio.Cancelled) else exc):
@@ -163,12 +164,12 @@ class InstanceWidget(QWidget):
         self.config = ParsecApp.get_main_window().config
 
         self.running_core_job = self.jobs_ctx.submit_job(
-            self.run_core_success,
-            self.run_core_error,
+            (self, "run_core_success"),
+            (self, "run_core_error"),
             _do_run_core,
             self.config,
             device,
-            self.run_core_ready,
+            (self, "run_core_ready"),
         )
 
     def on_run_core_ready(self, core, core_jobs_ctx):

--- a/parsec/core/gui/new_version.py
+++ b/parsec/core/gui/new_version.py
@@ -201,8 +201,8 @@ class CheckNewVersion(QDialog, Ui_NewVersionDialog):
         self.check_new_version_error.connect(self.on_check_new_version_error)
 
         self.version_job = self.jobs_ctx.submit_job(
-            self.check_new_version_success,
-            self.check_new_version_error,
+            (self, "check_new_version_success"),
+            (self, "check_new_version_error"),
             do_check_new_version,
             api_url=self.config.gui_check_version_api_url,
             allow_prerelease=self.config.gui_check_version_allow_pre_release,

--- a/parsec/core/gui/timestamped_workspace_widget.py
+++ b/parsec/core/gui/timestamped_workspace_widget.py
@@ -42,8 +42,8 @@ class TimestampedWorkspaceWidget(QWidget, Ui_TimestampedWorkspaceWidget):
         self.get_creation_timestamp_success.connect(self.on_success)
         self.get_creation_timestamp_error.connect(self.on_error)
         self.limits_job = self.jobs_ctx.submit_job(
-            self.get_creation_timestamp_success,
-            self.get_creation_timestamp_error,
+            (self, "get_creation_timestamp_success"),
+            (self, "get_creation_timestamp_error"),
             _do_workspace_get_creation_timestamp,
             workspace_fs=workspace_fs,
         )

--- a/parsec/core/gui/trio_jobs.py
+++ b/parsec/core/gui/trio_jobs.py
@@ -1,10 +1,12 @@
 # Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
 
 import threading
+from typing import List, Dict, Tuple, Callable, Any
 from inspect import iscoroutinefunction, signature
 from contextlib import asynccontextmanager
 import trio
 from structlog import get_logger
+from PyQt5.QtCore import QObject, pyqtBoundSignal
 
 from parsec.core.fs import FSError
 from parsec.core.mountpoint import MountpointError
@@ -12,6 +14,9 @@ from parsec.utils import split_multi_error
 
 
 logger = get_logger()
+
+
+JobResultSignal = Tuple[QObject, str]
 
 
 class JobResultError(Exception):
@@ -26,7 +31,23 @@ class JobSchedulerNotAvailable(Exception):
 
 
 class QtToTrioJob:
-    def __init__(self, fn, args, kwargs, on_success, on_error):
+    def __init__(
+        self,
+        fn: Callable,
+        args: List[Any],
+        kwargs: Dict[str, Any],
+        on_success: JobResultSignal,
+        on_error: JobResultSignal,
+    ):
+        # `pyqtBoundSignal` (i.e. `pyqtSignal` connected to a QObject instance) doesn't
+        # hold a strong reference on the QObject. Hence if the latter gets freed, calling
+        # the signal will result in a segfault !
+        # To avoid this, we should instead pass a tuple (<QObject instance>, <name of signal>).
+        assert all(not isinstance(a, pyqtBoundSignal) for a in args)
+        assert all(not isinstance(v, pyqtBoundSignal) for v in kwargs.values())
+        assert not isinstance(on_success, pyqtBoundSignal)
+        assert not isinstance(on_error, pyqtBoundSignal)
+
         self._on_success = on_success
         self._on_error = on_error
         self._fn = fn
@@ -124,7 +145,8 @@ class QtToTrioJob:
         self._done.set()
         signal = self._on_success if self.is_ok() else self._on_error
         if signal is not None:
-            signal.emit(self)
+            obj, name = signal
+            getattr(obj, name).emit(self)
 
     def cancel(self):
         self.cancel_scope.cancel()
@@ -188,7 +210,9 @@ class QtToTrioJobScheduler:
             self.nursery.start_soon(_throttled_execute, job)
         return job
 
-    def submit_job(self, on_success, on_error, fn, *args, **kwargs):
+    def submit_job(
+        self, on_success: JobResultSignal, on_error: JobResultSignal, fn: Callable, *args, **kwargs
+    ):
         job = QtToTrioJob(fn, args, kwargs, on_success, on_error)
         if self.nursery._closed:
             job.set_cancelled(JobSchedulerNotAvailable())

--- a/parsec/core/gui/users_widget.py
+++ b/parsec/core/gui/users_widget.py
@@ -328,8 +328,8 @@ class UsersWidget(QWidget, Ui_UsersWidget):
             return
 
         self.jobs_ctx.submit_job(
-            self.invite_user_success,
-            self.invite_user_error,
+            (self, "invite_user_success"),
+            (self, "invite_user_error"),
             _do_invite_user,
             core=self.core,
             email=user_email,
@@ -368,8 +368,8 @@ class UsersWidget(QWidget, Ui_UsersWidget):
         if r != _("TEXT_USER_INVITE_CANCEL_INVITE_ACCEPT"):
             return
         self.jobs_ctx.submit_job(
-            self.cancel_invitation_success,
-            self.cancel_invitation_error,
+            (self, "cancel_invitation_success"),
+            (self, "cancel_invitation_error"),
             _do_cancel_invitation,
             core=self.core,
             token=token,
@@ -424,8 +424,8 @@ class UsersWidget(QWidget, Ui_UsersWidget):
         if result != _("ACTION_USER_REVOCATION_CONFIRM"):
             return
         self.jobs_ctx.submit_job(
-            self.revoke_success,
-            self.revoke_error,
+            (self, "revoke_success"),
+            (self, "revoke_error"),
             _do_revoke_user,
             core=self.core,
             user_info=user_info,
@@ -573,8 +573,8 @@ class UsersWidget(QWidget, Ui_UsersWidget):
         self.spinner.show()
         pattern = self.line_edit_search.text()
         self.jobs_ctx.submit_job(
-            self.list_success,
-            self.list_error,
+            (self, "list_success"),
+            (self, "list_error"),
             _do_list_users_and_invitations,
             core=self.core,
             page=self._page,

--- a/parsec/core/gui/workspace_sharing_widget.py
+++ b/parsec/core/gui/workspace_sharing_widget.py
@@ -255,8 +255,8 @@ class WorkspaceSharingWidget(QWidget, Ui_WorkspaceSharingWidget):
         if sharing_widget:
             sharing_widget.set_status_updating()
         self.jobs_ctx.submit_job(
-            self.share_success,
-            self.share_error,
+            (self, "share_success"),
+            (self, "share_error"),
             _do_share_workspace,
             user_fs=self.user_fs,
             workspace_fs=self.workspace_fs,
@@ -355,8 +355,8 @@ class WorkspaceSharingWidget(QWidget, Ui_WorkspaceSharingWidget):
         self.spinner.show()
         self.widget_users.hide()
         self.jobs_ctx.submit_job(
-            self.get_users_success,
-            self.get_users_error,
+            (self, "get_users_success"),
+            (self, "get_users_error"),
             _do_get_users,
             core=self.core,
             workspace_fs=self.workspace_fs,

--- a/parsec/core/gui/workspaces_widget.py
+++ b/parsec/core/gui/workspaces_widget.py
@@ -1,12 +1,8 @@
 # Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
 
-from parsec.core.core_events import CoreEvent
-
-from PyQt5.QtCore import pyqtSignal, Qt, QTimer
+from PyQt5.QtCore import QObject, pyqtSignal, pyqtBoundSignal, Qt, QTimer
 from PyQt5.QtWidgets import QWidget, QLabel
-
-from parsec._parsec import DateTime, LocalDateTime
-
+from typing import Tuple
 from contextlib import contextmanager
 from structlog import get_logger
 from parsec.core.types import (
@@ -16,6 +12,9 @@ from parsec.core.types import (
     BackendOrganizationFileLinkAddr,
     WorkspaceRole,
 )
+
+from parsec._parsec import DateTime, LocalDateTime
+from parsec.core.core_events import CoreEvent
 from parsec.core.backend_connection import BackendNotAvailable
 from parsec.core.fs import (
     FsPath,
@@ -391,8 +390,8 @@ class WorkspacesWidget(QWidget, Ui_WorkspacesWidget):
             if button.is_owner:
                 try:
                     self.jobs_ctx.submit_job(
-                        self.reencryption_needs_success,
-                        self.reencryption_needs_error,
+                        (self, "reencryption_needs_success"),
+                        (self, "reencryption_needs_error"),
                         _get_reencryption_needs,
                         workspace_fs=workspace_fs,
                     )
@@ -497,8 +496,8 @@ class WorkspacesWidget(QWidget, Ui_WorkspacesWidget):
         if token in workspace_name:
             workspace_name, *_ = workspace_name.split(token)
             self.jobs_ctx.submit_job(
-                self.ignore_success,
-                self.ignore_error,
+                (self, "ignore_success"),
+                (self, "ignore_error"),
                 _do_workspace_rename,
                 core=self.core,
                 workspace_id=workspace_fs.workspace_id,
@@ -529,7 +528,10 @@ class WorkspacesWidget(QWidget, Ui_WorkspacesWidget):
                 else None,
             )
             self.jobs_ctx.submit_job(
-                self.file_open_success, self.file_open_error, desktop.open_files_job, [path]
+                (self, "file_open_success"),
+                (self, "file_open_error"),
+                desktop.open_files_job,
+                [path],
             )
         except MountpointNotMounted:
             # The mountpoint has been umounted in our back, nothing left to do
@@ -563,8 +565,8 @@ class WorkspacesWidget(QWidget, Ui_WorkspacesWidget):
         # the mount_success signal is not connected to anything but
         # is being kept for potential testing purposes
         self.jobs_ctx.submit_job(
-            self.mount_success,
-            self.mount_error,
+            (self, "mount_success"),
+            (self, "mount_error"),
             _do_workspace_mount,
             core=self.core,
             workspace_id=workspace_id,
@@ -577,8 +579,8 @@ class WorkspacesWidget(QWidget, Ui_WorkspacesWidget):
         # the unmount_success signal is not connected to anything but
         # is being kept for potential testing purposes
         self.jobs_ctx.submit_job(
-            self.unmount_success,
-            self.unmount_error,
+            (self, "unmount_success"),
+            (self, "unmount_error"),
             _do_workspace_unmount,
             core=self.core,
             workspace_id=workspace_id,
@@ -626,8 +628,8 @@ class WorkspacesWidget(QWidget, Ui_WorkspacesWidget):
         if not new_name:
             return
         self.jobs_ctx.submit_job(
-            self.rename_success,
-            self.rename_error,
+            (self, "rename_success"),
+            (self, "rename_error"),
             _do_workspace_rename,
             core=self.core,
             workspace_id=workspace_button.workspace_fs.workspace_id,
@@ -686,7 +688,10 @@ class WorkspacesWidget(QWidget, Ui_WorkspacesWidget):
             except FSError as exc:
                 raise JobResultError(ret=workspace_id, status="fs-error", origin=exc)
 
-        async def _reencrypt(on_progress, workspace_id):
+        async def _reencrypt(on_progress: Tuple[QObject, str], workspace_id):
+            on_progress: pyqtBoundSignal = getattr(
+                on_progress[0], on_progress[1]
+            )  # Retreive the signal
             with _handle_fs_errors():
                 if reencryption_already_in_progress:
                     job = await self.core.user_fs.workspace_continue_reencryption(workspace_id)
@@ -707,10 +712,10 @@ class WorkspacesWidget(QWidget, Ui_WorkspacesWidget):
         workspace_button.reencrypting = 1, 0
 
         self.jobs_ctx.submit_job(
-            self.workspace_reencryption_success,
-            self.workspace_reencryption_error,
+            (self, "workspace_reencryption_success"),
+            (self, "workspace_reencryption_error"),
             _reencrypt,
-            on_progress=self.workspace_reencryption_progress,
+            on_progress=(self, "workspace_reencryption_progress"),
             workspace_id=workspace_id,
         )
 
@@ -763,8 +768,8 @@ class WorkspacesWidget(QWidget, Ui_WorkspacesWidget):
         if not workspace_name:
             return
         self.jobs_ctx.submit_job(
-            self.create_success,
-            self.create_error,
+            (self, "create_success"),
+            (self, "create_error"),
             _do_workspace_create,
             core=self.core,
             workspace_name=workspace_name,
@@ -780,8 +785,8 @@ class WorkspacesWidget(QWidget, Ui_WorkspacesWidget):
         self.jobs_ctx.submit_throttled_job(
             "workspace_widget.list_workspaces",
             self.REFRESH_WORKSPACES_LIST_DELAY,
-            self.list_success,
-            self.list_error,
+            (self, "list_success"),
+            (self, "list_error"),
             _do_workspace_list,
             core=self.core,
         )

